### ISSUE TITLE
Reading: A Relational Memory-based Embedding Model for Triple Classification and Search Personalization

### ISSUE THREAD

**Issue by a1da4** [a1da4/paper-survey#149]
## 0. Paper
@inproceedings{nguyen-etal-2020-relational,
    title = "A Relational Memory-based Embedding Model for Triple Classification and Search Personalization",
    author = "Nguyen, Dai Quoc  and
      Nguyen, Tu  and
      Phung, Dinh",
    booktitle = "Proceedings of the 58th Annual Meeting of the Association for Computational Linguistics",
    month = jul,
    year = "2020",
    address = "Online",
    publisher = "Association for Computational Linguistics",
    url = "https://www.aclweb.org/anthology/2020.acl-main.313",
    doi = "10.18653/v1/2020.acl-main.313",
    pages = "3429--3435",
    abstract = "Knowledge graph embedding methods often suffer from a limitation of memorizing valid triples to predict new ones for triple classification and search personalization problems. To this end, we introduce a novel embedding model, named R-MeN, that explores a relational memory network to encode potential dependencies in relationship triples. R-MeN considers each triple as a sequence of 3 input vectors that recurrently interact with a memory using a transformer self-attention mechanism. Thus R-MeN encodes new information from interactions between the memory and each input vector to return a corresponding vector. Consequently, R-MeN feeds these 3 returned vectors to a convolutional neural network-based decoder to produce a scalar score for the triple. Experimental results show that our proposed R-MeN obtains state-of-the-art results on SEARCH17 for the search personalization task, and on WN11 and FB13 for the triple classification task.",
}
## 1. What is it?
> search personalization aims to re-rank the relevant documents returned by a user-oriented search system given a query.

## 2. What is amazing compared to previous works?
They proposed an R-MeN model from the relational memory network.
Their method achieves state-of-the-art performance in the SEARCH17 dataset. 

## 3. Where is the key to technologies and techniques?
<img width="465" alt="スクリーンショット 2021-03-01 2 00 07" src="https://user-images.githubusercontent.com/45454055/109426599-daeec780-7a31-11eb-834c-147a989e8dac.png">

From three vectors (subject, relation, and object), the score is obtained by CNN-based decoder.
<img width="523" alt="スクリーンショット 2021-03-01 2 12 42" src="https://user-images.githubusercontent.com/45454055/109426984-9b28df80-7a33-11eb-9fa4-5d2a0cba2b54.png">

This model is obtain by minimizing the loss function:
<img width="523" alt="スクリーンショット 2021-03-01 2 14 41" src="https://user-images.githubusercontent.com/45454055/109427039-e3e09880-7a33-11eb-88de-e66e5194d05e.png">

They define (subject, relation and object) as (query, user, and document) for the search personalization task.

### 4.1 Triple classification
<img width="523" alt="スクリーンショット 2021-03-01 2 23 38" src="https://user-images.githubusercontent.com/45454055/109427311-248ce180-7a35-11eb-9e43-020cf6f5b4f0.png">

### 4.2 Search personalization
<img width="523" alt="スクリーンショット 2021-03-01 2 24 06" src="https://user-images.githubusercontent.com/45454055/109427322-33739400-7a35-11eb-8389-cf1cde532db3.png">
